cms/djangoapps/models/settings/course_details.py

```diff
@@ -54,6 +54,8 @@ def __init__(self, org, course_id, run):
             '50'
         )  # minimum passing score for entrance exam content module/tree,
         self.has_cert_config = None  # course has active certificate configuration
+        if settings.FEATURES.get('ENABLE_SELF_PACED_COURSES'):
+            self.self_paced = None
```

Comment:
Is there any harm in recording this data always, and just preventing display if the feature is disabled? That would simplify the number of checks done in the code